packages/devtools_app/lib/src/console.dart

```diff
@@ -53,6 +53,45 @@ class Console extends StatelessWidget {
   }
 }
 
+/// Renders a _ConsoleText widget with ConsoleControls overlaid on the
+/// top-right corner.
+class ConsoleUsingTextWidget extends StatelessWidget {
+  const ConsoleUsingTextWidget({
+    this.controls,
+    @required this.lines,
+    this.title,
+  }) : super();
+
+  final Widget title;
+  final List<Widget> controls;
+  final List<String> lines;
+
+  String get textContent => lines.join('\n');
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        if (title != null) title,
+        Expanded(
+          child: Material(
+            child: Stack(
+              fit: StackFit.expand,
+              children: [
+                _ConsoleText(text: textContent),
```

Comment:
why isn't this a list view where each element is a line?

Reply:
This was in order to get line wrapping behavior in the logging page's detail area. The flutter error's sent for image overage had very long lines and some of the error message was lost.

I can play around w/ how we display json in that view instead - I think the long line had some newlines embedded in it that we weren't respecting (that were json escaped). I don't think the details area gets very long generally, so I wasn't worried about virtualizing the display.

Reply:
Ok. Virtualizing a list view in Flutter is trivial so it should just be the same lines of code but I agree the performance won't matter in this case unless we are displaying a really large block of text.